pkg/targetallocator/annotations.go

```diff
@@ -35,15 +35,21 @@ func Annotations(instance v1alpha1.OpenTelemetryCollector) map[string]string {
 
 	configMap, err := ConfigMap(instance)
 	if err == nil {
-		annotations[configMapHashAnnotationKey] = getConfigMapSHA(configMap)
+		cmHash := getConfigMapSHA(configMap)
+		if cmHash != "" {
+			annotations[configMapHashAnnotationKey] = getConfigMapSHA(configMap)
+		}
 	}
 
 	return annotations
 }
 
 // getConfigMapSHA returns the hash of the content of the TA ConfigMap.
 func getConfigMapSHA(configMap v1.ConfigMap) string {
-	configBytes := configMap.BinaryData[targetAllocatorFilename]
```

Comment:
this is the crux of the change, yeah?

Reply:
Yes. In my mind, I think I thought BinaryData was an alias for Data with a different type. I blame spending too many years writing Python.